### PR TITLE
New features

### PR DESCRIPTION
**Pull request by fdarricau**
This pull request is intended to bring in some new features:
- Polynomials::operator +=, for easier use (especially in loops)
- BSpline<N>::toPolynomials(), based on the CubicBSpline implementation
- Updating the matplotlib printing tool to allow the use of a BSpline of any dimension.

Warning: because there are still differences between the BSpline and the CubicBSpline (especially because of the different type naming), the implementation of the different plot_spline methods can't be completely merged in a single one.
There will have to be some work done there to avoid code duplication, once the basic differences between the CubicBSpline and BSpline will be no more.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -118,7 +118,7 @@ ROBOPTIM_TRAJECTORY_TEST(cubic-b-spline-gradient)
 ROBOPTIM_TRAJECTORY_TEST(b-spline)
 ROBOPTIM_TRAJECTORY_TEST(constrained-b-spline)
 ROBOPTIM_TRAJECTORY_TEST(spline-length)
-ROBOPTIM_TRAJECTORY_TEST(cubic-b-spline-matplotlib)
+ROBOPTIM_TRAJECTORY_TEST(b-spline-matplotlib)
```
Comment (bchretien) [inline tests/CMakeLists.txt]: What was the reason for removing `cubic-b-spline-matplotlib`? Why not keep it?

File: tests/b-spline-matplotlib.cc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        << plot_spline (*spline3, step)
+        << title ("BSpline, with P4 = {2.5, 15}")
+	);
+
+
+  std::cout << output->str() << std::endl;
+  BOOST_CHECK (output->match_pattern ());
+}
+
+BOOST_AUTO_TEST_CASE (trajectory_cubic_b_spline_matplotlib)
```
Comment (bchretien) [inline tests/b-spline-matplotlib.cc]: Although this is arguable, I'd prefer to have 2 different test files for `BSpline` and `CubicBSpline`.

Comment (fdarricau) [inline tests/b-spline-matplotlib.cc]: Understood

**Comment by bchretien** [comment]
:+1: LGTM

**Pull request merged**